patches/0132-ffmpeg-raisr-filter.patch

```diff
@@ -0,0 +1,443 @@
+   --enable-libjack         enable JACK audio sound server [no]
+   --enable-libjxl          enable JPEG XL de/encoding via libjxl [no]
+   --enable-libklvanc       enable Kernel Labs VANC processing [no]
+@@ -1871,6 +1872,7 @@ EXTERNAL_LIBRARY_LIST="
+     libgsm
+     libiec61883
+     libilbc
++    libipp
+     libjack
+     libjxl
+     libklvanc
+@@ -3822,6 +3824,7 @@ prewitt_opencl_filter_deps="opencl"
+ procamp_vaapi_filter_deps="vaapi"
+ program_opencl_filter_deps="opencl"
+ pullup_filter_deps="gpl"
++raisr_filter_deps="libipp"
+ remap_opencl_filter_deps="opencl"
+ removelogo_filter_deps="avcodec avformat swscale"
+ repeatfields_filter_deps="gpl"
+@@ -6860,6 +6863,18 @@ enabled libopus           && {
+         require_pkg_config libopus opus opus_multistream.h opus_multistream_surround_encoder_create
+     }
+ }
++
++if enabled libipp; then
++   ipp_header_for_check='ippcore.h'
++   case $target_os in
++       mingw32*|mingw64*)
++           ipp_header_for_check='_mingw.h ippcore.h'
++           ;;
++   esac
++   check_lib libipp "$ipp_header_for_check" ippInit -Wl,--start-group -lippi -lipps -lippcore -lippvm -Wl,--end-group ||
++   die "ERROR: Intel IPP not found"
++fi
++
+ enabled libplacebo        && require_pkg_config libplacebo "libplacebo >= 4.192.0" libplacebo/vulkan.h pl_vulkan_create
+ enabled libpulse          && require_pkg_config libpulse libpulse pulse/pulseaudio.h pa_context_new
+ enabled librabbitmq       && require_pkg_config librabbitmq "librabbitmq >= 0.7.1" amqp.h amqp_new_connection
+diff --git a/libavfilter/Makefile b/libavfilter/Makefile
+index 8a365c038b..541c2bdfe5 100644
+--- a/libavfilter/Makefile
++++ b/libavfilter/Makefile
+@@ -440,6 +440,7 @@ OBJS-$(CONFIG_PSNR_FILTER)                   += vf_psnr.o framesync.o
+ OBJS-$(CONFIG_PULLUP_FILTER)                 += vf_pullup.o
+ OBJS-$(CONFIG_QP_FILTER)                     += vf_qp.o
+ OBJS-$(CONFIG_QUIRC_FILTER)                  += vf_quirc.o
++OBJS-$(CONFIG_RAISR_FILTER)                  += vf_raisr.o
+ OBJS-$(CONFIG_RANDOM_FILTER)                 += vf_random.o
+ OBJS-$(CONFIG_READEIA608_FILTER)             += vf_readeia608.o
+ OBJS-$(CONFIG_READVITC_FILTER)               += vf_readvitc.o
+diff --git a/libavfilter/allfilters.c b/libavfilter/allfilters.c
+index 637bfae980..13abd38dd5 100644
+--- a/libavfilter/allfilters.c
++++ b/libavfilter/allfilters.c
+@@ -415,6 +415,7 @@ extern const AVFilter ff_vf_pullup;
+ extern const AVFilter ff_vf_qp;
+ extern const AVFilter ff_vf_qrencode;
+ extern const AVFilter ff_vf_quirc;
++extern const AVFilter ff_vf_raisr;
+ extern const AVFilter ff_vf_random;
+ extern const AVFilter ff_vf_readeia608;
+ extern const AVFilter ff_vf_readvitc;
+diff --git a/libavfilter/vf_raisr.c b/libavfilter/vf_raisr.c
+new file mode 100644
+index 0000000000..854703e790
+--- /dev/null
++++ b/libavfilter/vf_raisr.c
+@@ -0,0 +1,344 @@
++/*
++ * Intel Library for Video Super Resolution ffmpeg plugin
```

Comment:
Is this needed ?

Reply:
We need it. Maybe some server user want to use cpu version raisr.